src/librbd/AioImageRequest.cc

```diff
@@ -382,6 +390,107 @@ void AioImageWrite::update_stats(size_t length) {
   m_image_ctx.perfcounter->inc(l_librbd_wr_bytes, length);
 }
 
+void AioImageWriteSame::assemble_extent(const ObjectExtent &object_extent,
+                                    bufferlist *bl) {
+  if (object_extent.offset % m_data_len == 0
```

Comment:
Hard to say since we don't enforce a block size within librbd. Plus, if "fancy" striping is enabled, we can get some odd object extent mappings. I would just update this to support the potential for substrings, odd layouts, etc.